src/relations/peers.py

```diff
@@ -198,36 +150,17 @@ def _on_changed(self, event: HookEvent):
             - If pgbouncer container is unavailable.
         """
         self.unit_databag.update({ADDRESS_KEY: self.charm.unit_pod_hostname})
-        self.charm.update_client_connection_info()
 
         if self.charm.unit.is_leader():
-            try:
-                cfg = self.charm.read_pgb_config()
-            except FileNotFoundError:
-                # If there's no config, the charm start hook hasn't fired yet, so defer until it's
-                # available.
-                event.defer()
-                return
-
             self.update_leader()
-            return
-
-        if cfg := self.get_cfg():
-            self.charm.render_pgb_config(PgbConfig(cfg))
 
-        if auth_file := self.charm.get_secret(APP_SCOPE, AUTH_FILE_DATABAG_KEY):
-            self.charm.render_auth_file(auth_file)
+        if not self.charm.is_container_ready:
+            logger.debug("_on_peer_changed defer: container unavailable")
+            event.defer()
+            return
 
-        if cfg is not None or auth_file is not None:
-            try:
-                # raises an error if this is fired before on_pebble_ready.
-                self.charm.reload_pgbouncer()
-                self.charm.toggle_monitoring_layer(self.charm.backend.ready)
-            except (ConnectionError, ChangeError):
-                logger.error(
-                    "failed to reload pgbouncer - deferring change_event and waiting for pebble."
-                )
-                event.defer()
+        self.charm.render_pgb_config(reload_pgbouncer=True)
+        self.charm.toggle_monitoring_layer(self.charm.backend.ready)
```

Comment:
Most of the cfg updates and restarts will happen here, when `pgb_dbs_config` gets updated.